### PR TITLE
Adding useOrders hook

### PR DESCRIPTION
**Pull request by alfetopito**
Part of #208 

---

New hook to get user orders

### Note:
Build failed because I split changes into multiple PRs.
Following PR build works: #287

### CODE REVIEW

File: src/hooks/useOrders.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+import useSafeState from './useSafeState'
+import { exchangeApi } from 'api'
+import { useEffect } from 'react'
+
+export function useOrders(): AuctionElement[] {
+  const { userAddress } = useWalletConnection()
+  const [orders, setOrders] = useSafeState<AuctionElement[]>([])
+
+  useEffect(() => {
+    exchangeApi.getOrders(userAddress as string).then(_orders => {
```
Comment (Velenir) [inline src/hooks/useOrders.ts]:
```suggestion
    userAddress && exchangeApi.getOrders(userAddress).then(setOrders)
```

File: src/components/OrdersWidget/index.tsx [hunk 2]
```diff
@@ -131,6 +132,9 @@ const OrdersWidget: React.FC = () => {
   }, [])
   // end temporary layout vars
 
+  const orders = useOrders()
+  const noOrders = orders.length === 0
```
Comment (Velenir) [inline src/components/OrdersWidget/index.tsx]: Unused breaks, TS build

**Comment by Velenir** [comment]
`contract.methods.getEncodedUserOrders` thows
```
Error: invalid address (arg="user", coderType="address", value=undefined)
Error: Returned values aren't valid, did it run Out of Gas? You might also see this error if you are not using the correct ABI for the contract you are retrieving data from, requesting data from a block number that does not exist, or querying a node which is not fully synced.
```

Need to update contract address

**Comment by alfetopito** [comment]
> `contract.methods.getEncodedUserOrders` thows
> 
> ```
> Error: invalid address (arg="user", coderType="address", value=undefined)
> Error: Returned values aren't valid, did it run Out of Gas? You might also see this error if you are not using the correct ABI for the contract you are retrieving data from, requesting data from a block number that does not exist, or querying a node which is not fully synced.
> ```
> 
> Need to update contract address

We also need to update the contract and exchangeApi for this task https://github.com/gnosis/dex-react/issues/228

I think we could do it all at once

File: src/hooks/useOrders.ts [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+import { exchangeApi } from 'api'
+import { useEffect } from 'react'
+
+export function useOrders(): AuctionElement[] {
+  const { userAddress } = useWalletConnection()
+  const [orders, setOrders] = useSafeState<AuctionElement[]>([])
+
+  useEffect(() => {
+    userAddress && exchangeApi.getOrders(userAddress).then(setOrders)
+      setOrders(_orders)
```
Comment (Velenir) [inline src/hooks/useOrders.ts]: You just committed my suggestion ? :smile: Right here in github?

Multiline suggestions aren't here yet, but are on the github roadmap

Comment (alfetopito) [reply src/hooks/useOrders.ts]: I did! Though it was ready to go, no?

Comment (alfetopito) [reply src/hooks/useOrders.ts]: Ok now I see it. Fixed in latest revision :)

**Comment by W3stside** [comment]
APPROVED!

**Pull request merged**